docs_src/general/EdgeX_CN.md

```diff
@@ -0,0 +1,153 @@
+
+### CN Ingredients Not In EdgeX (and why)
+
+**Kubernetes**
+
+EdgeX provides example Helm Charts to assist adopters that want to run EdgeX in a Kubernetes environment.  However, EdgeX was not designed to fully operate in a multi-cluster environment and take advantage of a full K8s environment.  Our example Helm Charts, for example, allow a single instance of each EdgeX service to be deployed/orchestrated and monitored, but it would not allow K8s to fully manage and scale EdgeX services.  Why?  First and foremost Kubernetes is large compared to the resource constraints of some edge platforms.  While smaller Kubernetes environments are being developed for the edge (see *Futures* below), a whole host of challenges such as resource constraints, environment, infrastructure, etc. (as mentioned under *Containers* above) may not allow K8s to operate at the edge.  Kubernetes is, for the most part, about the ability to load balance, distribute traffic, and scale (up or down) workloads so that an application remains stable.  But on an edge platform, where would Kubernetes find the resources to balance and distribute and scale?  Because edge nodes are static and often times physically connected to the sensors they collect data from, there is not the means to grow and/or shift the workloads.  Portions of EdgeX might be able to scale up or down (those not physically tied to an edge sensor), but the platform as a whole is often rooted to the physical world it is connected to.
+
+**Serverless Functions**
+
+EdgeX is not built on a serverless execution model.  Unlike the compute and infrastructure resources of the cloud (which can almost be thought of as infinitely available and scaled up or down as needed), edge compute resources and infrastructure are not scaled up or down based on demand.  An edge gateway, running on a light pole of a smart city for example, is not dynamic.  The gateway must be provisioned based on the expected highest demand of that platform.  The workload on the edge gateway must operate within those resource constraints.  EdgeX is designed to operate in some of the smaller of the static, resource constrained environments.
```

Comment:
The idea of serverless and function-based need to be decoupled.  The idea of "serverless" at the edge is ridiculous - you are deploying physical nodes.  In the cloud, serverless means that you don't have the patch the VMs that are running your code because the CSP hides all of that from you.   You can get function-based applications using a toolkit like OpenFaaS, that is container-based under-the-hood.  Is there anything preventing most of the core of EdgeX from running as a FaaS architeture?  Probably not.

Reply:
@bnevis-i, We evaluated `OpenFaaS` at the start of App Service WG and found it was not use able for EdgeX at the time and we went with our functions pipeline approach. Have not revisited since  `OpenFaaS` .

Reply:
I am not suggesting in this section that EdgeX should be or is either serverless or function-based.  I am merely stating the "ingredients" that CNCF says applications must adhere to in order to be cloud native - and why EdgeX doesn't adhere to these (thus adding to the explanation of why EdgeX is not cloud native and never will be).  So we are in agreement here - serverless at the edge is not a fit.  And while much is possible, it is worth stating that EdgeX is not just "functions" today.